src/agent/common.ts

```diff
@@ -468,6 +455,24 @@ function createMaskFunction(jobEnvironment: agentifm.JobEnvironment): Replacemen
             }
         });
 
+        endpoints.forEach((endpoint: agentifm.ServiceEndpoint) => {
+            if (endpoint.authorization != null && endpoint.authorization['parameters'] != null) {
+                var keys = Object.keys(endpoint.authorization['parameters']);
+                keys.forEach((key: string) => {
+                    var valueToReplace: string = endpoint.authorization['parameters'][key];
```

Comment:
We should check to see whether valueToReplace is empty.  Otherwise, I think we'll create a zero-length ReplacementPosition for every character in the string.   "abcd".indexOf("") === 0